android/build.gradle

```diff
@@ -161,6 +161,7 @@ android {
     buildConfigField 'String', 'SUPPORT_MAIL', '"android@organicmaps.app"'
     buildConfigField 'String', 'REVIEW_URL', '"https://organicmaps.app/donate/"'
     buildConfigField 'int', 'RATING_THRESHOLD', '5'
+    resConfigs "en", "ar", "be", "bg", "ca", "cs", "da", "de", "el", "en_GB", "es", "es_MX", "eu", "fa", "fi", "fr", "fr_CA", "hu", "in", "it", "iw", "ja", "ko", "mr", "nb", "nl", "pl", "pt", "pt_BR", "ro", "ru", "sk", "sv", "sw", "th", "tr", "uk", "vi", "zh", "zh_HK", "zh_MO", "zh_TW"
```

Comment:
zh_TW or zh-rTW?

Reply:
zh_TW take in zh-rTW during compilation.
I have use per-app language and strings in values_fr-rCA.xml are available in app.

Reply:
OK, I tested that `en-rGB` works

Reply:
1. Please use single quotes for constants without vars inside them.
2. Is it possible to load them from a config file?
3. Is there a way to automatically calculate and set these languages based on our real translations?
4. Please update the documentation so that when a new language is added, this line should be updated too.

Reply:
> 1. Please use single quotes for constants without vars inside them.

Yes :)

>  2. Is it possible to load them from a config file?

Yes, we can add resConfig in [gradle.properties](https://github.com/organicmaps/organicmaps/blob/master/android/gradle.properties) Check [this](https://medium.com/livefront/improve-android-build-times-by-using-resconfigs-99177a3c6d90)

> 3. Is there a way to automatically calculate and set these languages based on our real translations?

Not, you need to specify which language do you want integrate in Android build

> 4. Please update the documentation so that when a new language is added, this line should be updated too.

Do you want i add new section in [TRANSLATIONS.md](https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md) ? Or just specify a new line in Translation files section with link to build.gradle with description `List of languages integrate in Android build`

Reply:
> > 1. Please use single quotes for constants without vars inside them.
> 
> Yes :)

Thanks! This is not documented, but here is a strong belief that single quotes are easier to read than double quota, which is a case in Bash. I filed a follow-up ticket to research https://github.com/organicmaps/organicmaps/issues/3684.

> 
> > 2. Is it possible to load them from a config file?
> 
> Yes, we can add resConfig in [gradle.properties](https://github.com/organicmaps/organicmaps/blob/master/android/gradle.properties) Check [this](https://medium.com/livefront/improve-android-build-times-by-using-resconfigs-99177a3c6d90)

I don't see any difference between having this list in `gradle.properties` vs `build.gradle`. It doesn't change things significantly. Both options look like the same for me. 

@biodranik do you have any better ideas? 

> 
> > 3. Is there a way to automatically calculate and set these languages based on our real translations?
> 
> Not, you need to specify which language do you want integrate in Android build
> 

Generating this list dynamically based on `values-*` directory sounds complicated to me. It is just not worth it.

> > 4. Please update the documentation so that when a new language is added, this line should be updated too.
> 
> Do you want i add new section in [TRANSLATIONS.md](https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md) ? Or just specify a new line in Translation files section with link to build.gradle with description `List of languages integrate in Android build`

I don't see a good place where you can add a comment in translation files (string.txt). TRANSLATIONS.md should be fine.

Reply:
I have a better idea. If languages can be loaded from file, let's update tools/unix/generate_localizations.sh to autogenerate this file with the list of all supported languages, for example, in strings.txt.

Reply:
We don't add new languages every day. Can we keep it simple?

Reply:
It is very easy to forget it. With the help of the community, I expect many languages to be added. Let's do it properly from the start. Once and for all.

Reply:
This one-liner can be inserted into generate_localizations.sh to autogenerate the include config for android:
```bash
cat data/strings/strings.txt| sed -nE "s/ +([a-zA-Z]{2}(-[a-zA-Z]{2,})?) = .*$/'\1',/p" | sort -u | tr '\n' ' ' | sed -E 's/, $//'
'ar', 'be', 'bg', 'ca', 'cs', 'da', 'de', 'el', 'en', 'en-GB', 'es', 'es-MX', 'et', 'eu', 'fa', 'fi', 'fr', 'fr-CA', 'he', 'hi', 'hu', 'id', 'in', 'it', 'ja', 'ko', 'mr', 'nb', 'ne', 'nl', 'pl', 'pt', 'pt-BR', 'ro', 'ru', 'sk', 'sv', 'sw', 'th', 'tr', 'uk', 'vi', 'zh-Hans', 'zh-Hant'
```

Reply:
Can you give more information about what doing these commands, to better understand what i need to add in build.gradle.
Is it ok, if i add your commands just in end file before rm in .generate_localizations.sh ?

Reply:
The first line is the command.
The second line is its output.

You need to include a file with these langs into build.gradle instead of hard-coding languages there, and generate this file automatically from the mentioned script.

Reply:
- List of languages is include in gradle.properties
- resConfig call gradle.properties (It's more simple solution, i have try with generate txt file, but i didn't have find how import content of file and use in resConfig)
List of languages is more longer and when i try to build, compilation fail, can you check ?
More logs here :https://github.com/Jean-BaptisteC/organicmaps/actions/runs/3450378278/jobs/5758934308